virttest/remote_commander/remote_runner.py

```diff
@@ -746,7 +745,9 @@ def python_file_run_with_helper(self, test_path):
         :return: module.run().return
         """
         assert os.path.isfile(test_path)
-        module = imp.load_source('ext_test', test_path)
+        sys.path.insert(0, test_path)
+        module = importlib.import_module('ext_test')
+        sys.path.pop(0)
```

Comment:
For the replacement of `imp.load_source`, looks at least we have to take care about these points:
```python
$ cat /tmp/t.py
def hi():
    print('hi')

# a normal importing case
$ python
Python 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 11.2.1 20220127 (Red Hat 11.2.1-9)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> import imp
<stdin>:1: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
>>> t = imp.load_source('t', '/tmp/t.py')
>>> sys.modules['t']
<module 't' from '/tmp/t.py'>
>>> t.hi()
hi
```

1. `imp.load_source` requires the 2nd argument to be the file path of the given python file, which means for our new approach there is some processing work had to be done before passing it to `sys.path.insert`.
```python
>>> imp.load_source('t', '/tmp')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 719, in _load
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1016, in get_code
  File "/usr/lib64/python3.10/imp.py", line 158, in get_data
    return super().get_data(path)
  File "<frozen importlib._bootstrap_external>", line 1073, in get_data
IsADirectoryError: [Errno 21] Is a directory: '/tmp'
```
```python
>>> import sys
>>> import importlib
>>> sys.path.insert(0, '/tmp/t.py')
>>> t = importlib.import_module('t')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 't'
>>> sys.path.pop(0)
'/tmp/t.py'
>>> sys.path.insert(0, '/tmp')
>>> t = importlib.import_module('t')
>>> t.hi()
hi
```

2. the 1st argument of `imp.load_source` which is the name of the module doesn't have to be the same name to the python file, which is kind of tricky to the new approach.
```python
>>> foo = imp.load_source('foo', '/tmp/t.py')
>>> sys.modules['foo']
<module 'foo' from '/tmp/t.py'>
>>> foo.hi()
hi
```
```python
>>> sys.path.insert(0, '/tmp')
>>> importlib.import_module('foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'foo'
```

3. when the python file is not existed for whatever reason, the new approach could not give the expected result in the case that there was other module having the same module name stored in `sys.path`.
```python
>>> imp.load_source('os', '/tmp/nonexistfile.py')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/imp.py", line 170, in load_source
    module = _exec(spec, sys.modules[name])
  File "<frozen importlib._bootstrap>", line 619, in _exec
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1016, in get_code
  File "/usr/lib64/python3.10/imp.py", line 158, in get_data
    return super().get_data(path)
  File "<frozen importlib._bootstrap_external>", line 1073, in get_data
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/nonexistfile.py'
```
```python
>>> sys.path.insert(0, '/tmp')
>>> m = importlib.import_module('os')
>>> m.__file__
'/usr/lib64/python3.10/os.py'
>>> m.hi()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'os' has no attribute 'hi'
```

Reply:
Maybe we can could some inspiration from the two places below?

[1] https://github.com/python/cpython/blob/5397b5afc1f594dc0ba3b7743351d595e637bf24/Lib/imp.py#L166-L177
[2] https://docs.python.org/3.10/library/importlib.html?highlight=importlib#importing-a-source-file-directly